SudoScript.Core/Solver.cs

```diff
@@ -71,14 +72,96 @@ private static bool SolveRec(Board board, [NotNullWhen(true)] out Board? solvedB
         return false;
     }
 
+    /// <summary>
+    /// Finds a number of solutions to a given board.
+    /// </summary>
+    /// <param name="board"></param>
+    /// <param name="random">Whether the solved boards should be randomized.</param>
+    /// <param name="solutionCount">Maximum number of solutions that should be in the returned list. Zero means unlimited.</param>
+    /// <returns></returns>
+    public static List<Board>? FindSolutions(Board board, int limit = 0, bool random = false)
+    {
+        HashSet<Board> result = SolveRecAll(board, 0, limit, random);
+        // If a limit is specified and the result count exceeds it, return only the correct number of solutions.
+        if (limit != 0 && result.Count > limit)
+        {
+            return result.Take(limit).ToList();
```

Comment:
Is there a reason to take a limit here? Shouldn't SolveRecAll already give you the limit?

Reply:
It's the principle of having a bounded search. This will stop the recursion early if no more solutions are requested

Reply:
Then i think you should try to gain an understanding of that. A test like this is really error prone and probably not even needed.